### PR TITLE
upgpkg(x11/rofi): update to 1.7.4

### LINKED ISSUE #11547: Update rofi to 1.7.3
New sha256 : 25868b68ff98338c5421211fa2f8286f9c5b6f900b4bfb7c99e0fe64eb4db67b

build.sh
```
TERMUX_PKG_HOMEPAGE=https://github.com/DaveDavenport/rofi
TERMUX_PKG_DESCRIPTION="A window switcher, application launcher and dmenu replacement"
TERMUX_PKG_LICENSE="MIT"
TERMUX_PKG_MAINTAINER="Tristan Ross <spaceboyross@yandex.com>"
TERMUX_PKG_VERSION=1.7.3
TERMUX_PKG_REVISION=5
TERMUX_PKG_SRCURL="https://github.com/DaveDavenport/rofi/releases/download/$TERMUX_PKG_VERSION/rofi-$TERMUX_PKG_VERSION.tar.xz"
TERMUX_PKG_SHA256=25868b68ff98338c5421211fa2f8286f9c5b6f900b4bfb7c99e0fe64eb4db67b
TERMUX_PKG_DEPENDS="libandroid-glob, libandroid-shmem, libcairo, librsvg, libxkbcommon, startup-notification, xcb-util-wm, xcb-util-xrm"
TERMUX_PKG_BUILD_IN_SRC=true
TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--disable-check"

termux_step_pre_configure() {
	export LIBS="-landroid-glob -landroid-shmem"
	export GLIB_GENMARSHAL=glib-genmarshal
	export GOBJECT_QUERY=gobject-query
	export GLIB_MKENUMS=glib-mkenums
	export GLIB_COMPILE_RESOURCES=glib-compile-resources
}
```

### PR DESCRIPTION
**Pull request by T-Dynamos**
Closes #11547

### CODE REVIEW

File: x11-packages/rofi/build.sh [hunk 1]
```diff
@@ -2,10 +2,10 @@ TERMUX_PKG_HOMEPAGE=https://github.com/DaveDavenport/rofi
 TERMUX_PKG_DESCRIPTION="A window switcher, application launcher and dmenu replacement"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="Tristan Ross <spaceboyross@yandex.com>"
-TERMUX_PKG_VERSION=1.6.1
+TERMUX_PKG_VERSION=1.7.4
 TERMUX_PKG_REVISION=5
```
Comment (2096779623) [inline x11-packages/rofi/build.sh:6]: Remove `TERMUX_PKG_REVISION` .

**Review by 2096779623 — CHANGES_REQUESTED** [review]

**Comment by T-Dynamos** [comment]
Do I have to remove patch file?

**Comment by 2096779623** [comment]
> Do I have to remove patch file?

No, This file is now [here](https://github.com/davatorium/rofi/blob/master/source/modes/combi.c), you only need to remake the patch.

https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#submitting-new-packages-checklist

**Comment by T-Dynamos** [comment]
I am unable to understand the error
```
#define _TANDEM_SOURCE 1
#define __EXTENSIONS__ 1
#define ENABLE_DRUN 1
#define WINDOW_MODE 1
#define NK_XKBCOMMON_HAS_COMPOSE 1
#define NK_XKBCOMMON_HAS_CONSUMED2 1
#define GLIB_VERSION_MIN_REQUIRED (G_ENCODE_VERSION(2,40))
#define GLIB_VERSION_MAX_ALLOWED (G_ENCODE_VERSION(2,40))

configure: exit 1
Error: Process completed with exit code 1.
````
Please check if my patch file is right [source-modes-combi.c.patch](https://github.com/termux/termux-packages/blob/e701224e9da72f1a7423826cc4ae2b5ffb66c9d7/x11-packages/rofi/source-modes-combi.c.patch)

**Comment by 2096779623** [comment]
```bash
checking for glib-2.0 >= 2.36 xcb xcb-aux xcb-xkb xkbcommon xkbcommon-x11 xcb-ewmh xcb-icccm xcb-cursor xcb-randr xcb-xinerama... no
configure: error: Package requirements (glib-2.0 >= 2.36 xcb xcb-aux xcb-xkb xkbcommon xkbcommon-x11 xcb-ewmh xcb-icccm xcb-cursor xcb-randr xcb-xinerama) were not met:

No package 'xcb-cursor' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GW_XCB_INTERNAL_CFLAGS
and GW_XCB_INTERNAL_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by rofi configure 1.7.4, which was
generated by GNU Autoconf 2.71.  Invocation command line was

  $ /home/builder/.termux-build/rofi/src/configure --disable-dependency-tracking --prefix=/data/data/com.termux/files/usr --libdir=/data/data/com.termux/files/usr/lib --sbindir=/data/data/com.termux/files/usr/bin --disable-rpath --disable-rpath-hack --host=aarch64-linux-android --disable-check --disable-nls --enable-shared --enable-static --libexecdir=/data/data/com.termux/files/usr/libexec
```

**Comment by T-Dynamos** [comment]
Ohh So we need to add `xcb-cursor` first

**Review by 2096779623 — APPROVED** [review]

File: x11-packages/rofi/source-modes-combi.c.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
+diff '--color=auto' -uNr rofi-1.7.4/source/modes/combi.c rofi-1.7.4.mod/source/modes/combi.c
+--- rofi-1.7.4/source/modes/combi.c	2022-07-28 04:08:02.000000000 +0530
++++ rofi-1.7.4.mod/source/modes/combi.c	2022-08-19 10:27:45.192796288 +0530
+@@ -142,6 +142,17 @@
+     mode_set_private_data(sw, NULL);
+   }
+ }
++
++#if defined(__ANDROID_API__) && __ANDROID_API__ < 24
```
Comment (buttaface) [inline x11-packages/rofi/source-modes-combi.c.patch:9]: Do we need this? We only support API 24 and up now.

Comment (2096779623) [reply x11-packages/rofi/source-modes-combi.c.patch:9]: > Do we need this? We only support API 24 and up now.

Same idea, termux for Android 5 does not have X11 repo for the time being.

Comment (T-Dynamos) [reply x11-packages/rofi/source-modes-combi.c.patch:9]: So the patch isn't needed?

Comment (truboxl) [reply x11-packages/rofi/source-modes-combi.c.patch:9]: @sylirre do we still need this?

Comment (sylirre) [reply x11-packages/rofi/source-modes-combi.c.patch:9]: Remove it

**Pull request merged**